src/vs/platform/quickinput/browser/quickInputTree.ts

```diff
@@ -741,25 +741,6 @@ export class QuickInputTree extends Disposable {
 				indent: 0,
 				horizontalScrolling: false,
 				allowNonCollapsibleParents: true,
-				identityProvider: {
-					getId: element => {
-						// always prefer item over separator because if item is defined, it must be the main item type
-						const mainItem = element.item || element.separator;
-						if (mainItem === undefined) {
-							return '';
-						}
-						// always prefer a defined id if one was specified and use "label + description + detail" as a fallback
-						if (mainItem.id !== undefined) {
-							return mainItem.id;
-						}
-						let id = `label:${mainItem.label}`;
-						id += `$$description:${mainItem.description}`;
-						if (mainItem.type !== 'separator') {
-							id += `$$detail:${mainItem.detail}`;
-						}
-						return id;
-					},
-				},
```

Comment:
I still find it strange that we remove the identity provider here. Is there no way to get proper ids for each element?

Reply:
I don't think this can be done with an identity provider. When identity is equal, the tree keeps the 'old' object. This is not the behavior we want for quick pick.

For example, let's say Element A has a label of `foo`, no description, no detail, but a QP item button of `trash` and we're gonna `setChildren` with an Element of B that has the same label, description, detail, but a button of `plus`. This identity provider above would keep the `trash` button, but we want the `plus` button.

In all cases, we want the tree to have the 'new' object, and it looks like that's not the behavior of the tree when using an Identity Provider, so I don't think we want to use an Identity Provider.

Reply:
This is only because you're defining identity to be a function of appearance. I understand why, since QP API is like that.

But if the QP items were, say, files, then their identity would be a function of their URI. And their actions _visibility_ would be a function of context.

All of this is OK, it will make QP not be as performance at it could (since all elements will always be trashed across `setChildren` calls instead of being reused) but I doubt it will make a perceptual difference.